### PR TITLE
Fix component hiding

### PR DESCRIPTION
**Pull request by Robot8lover**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

Exchange instances of `hidden md:block` or `md:hidden` and similar CSS based hiding techniques for `v-if` statements using breakpoints. As mentioned in #864, this has the advantage of not rendering these components, which is better for performance and usually desirable unless the window is being resized. If accounting for resizing windows is preferred, though it is usually not the case, then `v-show` could be used instead.
I tried to keep any `display` property classes unless it was obvious they could be removed (e.g. block on a `div`). It is possible I reversed some of the logic due to a forgotten exclamation mark, which is one of the downsides of what I did. It may be better to explicitly write `greaterOrEqual` or `lesser` to avoid such errors. There were also some conditionally assigned hiding classes, which involve two boolean checks.

### Related issue

- #864

### CODE REVIEW

**Comment by andrewtavis** [comment]
Really appreciate all the work here, @Robot8lover! We have the dev sync on Saturday that you'd be more then welcome to :) If you can't make the sync, also let me know and I'd be happy to set up a call and answer any questions you might have!

I'll try to bring all the PRs in over the coming days 😊

**Review by andrewtavis — APPROVED** [review]
This change really is great, @Robot8lover 😊 Exactly what we needed, and was really easy to follow! :)

I added/changed a few things:
- I updated the style guide so that people know to follow these conventions
- I changed the names of instances of `useBreakpoint` per your suggestion to make it a bit more clear in the code that the conditional has to do with breakpoints
  - Rather than `medium` we now have `aboveMediumBP`

From my view the conditionals have all been applied correctly, but we can fix something if need be later on!

File: frontend/components/header/HeaderWebsite.vue [hunk 1]
```diff
@@ -10,7 +10,7 @@
     }"
   >
     <!-- MARK: Mobile Header -->
-    <div id="mobile-header" class="flex px-4 py-3 md:hidden">
+    <div v-if="!aboveMediumBP" class="flex px-4 py-3">
```
Comment (aasimsyed) [inline frontend/components/header/HeaderWebsite.vue:13]: removing this id breaks some tests

Comment (andrewtavis) [reply frontend/components/header/HeaderWebsite.vue:13]: This one's on me, @aasimsyed! Sorry about that!

**Pull request merged**